studies/models.py

```diff
@@ -776,6 +777,26 @@ def generate_videos_from_events(self):
         return Video.objects.bulk_create(video_objects)
 
 
+@receiver(post_save, sender=Response)
+def take_action_on_exp_data(sender, instance, created, **kwargs):
+    """Performs post-save actions based on the current frame.
+
+    For now, this just includes deleting videos for withdrawn videos.
+    """
+    response = instance  # Aliasing because instance is hooked as a kwarg.
+
+    if not response.sequence:
+        return
+
+    current_frame = response.sequence[-1]
+
+    if any(postfix in current_frame for postfix in VALID_EXIT_FRAME_POSTFIXES):
+        # check for withdrawal
+        withdrawal = response.exp_data[current_frame].get("withdrawal", False)
+        if withdrawal:
+            response.videos.delete()  # Should trigger a pre-delete hook.
```

Comment:
We should actually not include consent footage in the videos to be deleted (for now - leaving this possibility open in the future would be sensible)

Reply:
done